### ISSUE TITLE
Statusline integration

### ISSUE THREAD

**Issue by ghost** [autozimu/LanguageClient-neovim#742]
## Integrating into Statusline 
LanguageClient-neovim is great, it's extremely fast and I love that it is built on rust.
I use [Ale](https://github.com/w0rp/ale) for linting languages that don't have stable language servers, I've noticed their README has examples on how to integrate Ale into [Lightline](https://github.com/itchyny/lightline.vim)(My favorite Statusline) and for vim's builtin statusline(Hardecore Vimmer's Statusline)

## Solution
I think [Ale](https://github.com/w0rp/ale) has a very well filled out README and I'd like to see LanguageClient-neovim's README at the same level, but I'd especially like to see how to integrate it into my lightline status line without using a [plugin](https://github.com/Palpatineli/lightline-lsc-nvim) and into a normal status line so that people looking to make the new Airline | Lightline | Powerline alternative can have day 1 LanguageClient-neovim integration support

**Comment by jubnzv** [comment]
Another solution is to write [a small function](https://github.com/jubnzv/dotfiles/blob/master/vim/.config/nvim/init.vim#L773) in your `.vimrc` to display LanguageClient status:
```vim
augroup LanguageClient_config
  au!
  au User LanguageClientStarted call LSPUpdateStatus(1)
  au User LanguageClientStopped call LSPUpdateStatus(0)
augroup END
let g:lsp_status = 0
function! LSPUpdateStatus(status) abort
  let g:lsp_status = a:status
  call lightline#update()
endfunction
function! LightlineLSPStatus() abort
  return g:lsp_status == 1 ? 'LSP' : ''
endfunction
```

And add it to [lightline configuration](https://github.com/jubnzv/dotfiles/blob/master/vim/.config/nvim/init.vim#L326) like this:
```vim
let g:lightline = {
  \ 'component_expand': {
  \   'lsp_status': 'LightlineLSPStatus'
  \ },
  \ }
```

**Comment by Kotaro7750** [comment]
jubnz's way does not work when you edit two files including file which lsp is not working for. 
For example, assuming that you enabled cpp-languageserver, when you open cpp file and open txt file in another buffer, g:lsp_status remains 1 in txt file buffer.

**Comment by Kotaro7750** [comment]
Although I have not checked source code yet, a function like LanguageClient#isServerRunningForCurrentBuffer will solve this problem.

```vim
function! LightlineLSPStatus() abort
  return LanguageClient#isServerRunningForCurrentBuffer() ? 'LSP' : ''
endfunction
```

If current implementation support buffer-local language-server, that function can be easily implemented, but if not, we cannot recognize buffer which language-server is working for, so implementation will be difficult.

**Comment by martskins** [comment]
This is now solved by #1003 and the bufvar `b:LanguageClient_isServerRunning`
